src/vs/platform/terminal/node/terminalProcess.ts

```diff
@@ -187,6 +187,27 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 			return firstError;
 		}
 
+		let injection: IShellIntegrationConfigInjection | undefined;
+		if (this._options.shellIntegration) {
+			injection = getShellIntegrationInjection(this.shellLaunchConfig, this._options.shellIntegration);
+			if (!injection) {
+				this._logService.warn(`Shell integration cannot be enabled for executable "${this.shellLaunchConfig.executable}" and args`, this.shellLaunchConfig.args);
+			} else {
+				if (injection.envMixin) {
+					for (const [key, value] of Object.entries(injection.envMixin)) {
+						this._ptyOptions.env ||= {};
+						this._ptyOptions.env[key] = value;
+					}
+				}
+				if (injection.filesToCopy) {
+					for (const f of injection.filesToCopy) {
```

Comment:
maybe a comment about why this is happening IE for zsh?

Reply:
This is the generic part but I'll explain it in the inject function

Reply:
This is the generic part, it's fairly obvious when you look at references and has this comment so it's clear:

```
// Move .zshrc into $ZDOTDIR as the way to activate the script
```